qcodes/instrument_drivers/rohde_schwarz/ZNB.py

```diff
@@ -417,22 +423,37 @@ def __init__(self, name: str, address: str, init_s_params: bool=True, **kwargs)
                            get_cmd='OUTP1?',
                            set_cmd='OUTP1 {}',
                            val_mapping={True: '1\n', False: '0\n'})
-        self.add_function('reset', call_cmd='*RST')
-        self.add_function('tooltip_on', call_cmd='SYST:ERR:DISP ON')
-        self.add_function('tooltip_off', call_cmd='SYST:ERR:DISP OFF')
-        self.add_function('cont_meas_on', call_cmd='INIT:CONT:ALL ON')
-        self.add_function('cont_meas_off', call_cmd='INIT:CONT:ALL OFF')
-        self.add_function('update_display_once', call_cmd='SYST:DISP:UPD ONCE')
-        self.add_function('update_display_on', call_cmd='SYST:DISP:UPD ON')
-        self.add_function('update_display_off', call_cmd='SYST:DISP:UPD OFF')
-        self.add_function('display_sij_split', call_cmd='DISP:LAY GRID;:DISP:LAY:GRID {},{}'.format(
-            num_ports, num_ports))
-        self.add_function('display_single_window',
-                          call_cmd='DISP:LAY GRID;:DISP:LAY:GRID 1,1')
-        self.add_function('display_dual_window',
-                          call_cmd='DISP:LAY GRID;:DISP:LAY:GRID 2,1')
-        self.add_function('rf_off', call_cmd='OUTP1 OFF')
-        self.add_function('rf_on', call_cmd='OUTP1 ON')
+
+        self.add_parameter(name='tooltip',
+                           set_cmd='SYSTem:ERRor:DISPlay {}',
+                           get_cmd='SYSTem:ERRor:DISPlay?',
+                           get_parser=int,
+                           val_mapping={"ON": 1, "OFF": 0})
+
+        self.add_parameter(name='sweep_mode',
+                           label='Sweep mode',
+                           set_cmd='INITiate:CONTinuous:ALL {}',
+                           get_cmd='INITiate:CONTinuous:ALL?',
+                           val_mapping={'SINGLE': 0, 'CONTINUOUS': 1},
+                           get_parser=int)
+
+        self.add_parameter(name="display_updates",
+                           label="Display updates on instrument",
+                           set_cmd="SYSTem:DISPlay:UPDate {}",
+                           get_cmd="SYSTem:DISPlay:UPDate?",
+                           val_mapping={'ON': '1\n', 'OFF': '0\n',
+                                        'ONCE': 'ONCE\n'})
+
+        self.add_parameter(name='display_layout',
+                           label='Instrument display layout mode',
+                           set_cmd='DISPlay:LAYout {}',
+                           get_cmd='DISPlay:LAYout?',
+                           val_mapping={'LINEUP': 'LIN\n',
+                                        'STACK': 'STAC\n',
+                                        'HORIZONTAL': 'HOR\n',
+                                        'VERTICAL': 'VER\n',
+                                        'GRID': 'GRID\n'})
```

Comment:
really? `\n` here? the instrument manufacturer is crazy :)

Reply:
No, it all depends on what you set your termination character to. If it's set to `""`, then you manually deal with the newlines. You can also set it to `"\n"` and never see a newline in your life. I just followed the directions already given in the driver. (but I might change those directions)

Reply:
ok, i leave it up to you. this PR is about removing the functions, not about improving this driver.